### PR TITLE
schematelemetrycontroller: shift schedules in time, make tenant-ro

### LINKED ISSUE #85788: schematelemetry: reduce log burstiness
Presently, the schema telemetry samples the catalog thusly:
- once a week, on midnight on the first day of the week
- if more than 10k records in a snapshot, picks 10k records at random

The log collection infrastructure does not have infinite capacity and would instead rather see a steady trickle of messages rather than an infrequent burst. We need to determine exactly what hard constraints we're operating under, and flatten the curve of the schema telemetry event logging peaks somehow. Either log less stuff, or for less clusters, or at more random times, or emit the logs more gradually, or all of the above.

Jira issue: CRDB-18432

### PR DESCRIPTION
**Pull request by ajwerner**
This commit adds logic to rewrite certain special crontab specs to have the same frequency but a different point in that interval. This is useful to even out the burst of logs we'd surely recieve if all clusters logged at the same time. We don't anticipate that folks will set their own schedules, but for tenants, there is no good reason to let them control the schedules.

Usages of @hourly, @daily, or @weekly will be spread out accordingly.

The algorithm hashes the cluster ID to seed a random number generator which is used to determine the point in the interval for the schedule to occur.

Fixes #85788

Release note: None

### CODE REVIEW

File: pkg/sql/catalog/schematelemetry/schematelemetrycontroller/testdata/cron_rewrites [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+81cfb672-3ea8-11ed-b878-0242ac120002
+81cfb762-3ea8-11ed-b878-0242ac120002
+81cfb83e-3ea8-11ed-b878-0242ac120002
+81cfb92e-3ea8-11ed-b878-0242ac120002
+81cfbc4e-3ea8-11ed-b878-0242ac120002
+81cfbd48-3ea8-11ed-b878-0242ac120002
+81cfbe24-3ea8-11ed-b878-0242ac120002
+----
+                                      @hourly     @daily       @weekly
+440aaea9-06f5-43c1-94d4-fb2b57601ebd  22 * * * *  22 10 * * *  22 10 * * 5
```
Comment (chengxiong-ruan) [inline pkg/sql/catalog/schematelemetry/schematelemetrycontroller/testdata/cron_rewrites:29]: for `@weekly` why do we distribute on `month`? shouldn't we distribute on `day` instead?

Comment (ajwerner) [reply pkg/sql/catalog/schematelemetry/schematelemetrycontroller/testdata/cron_rewrites:29]: https://crontab.guru/#22_10_*_*_5

I don't think I distribute on month. The last column IIUC, is day of the week.

Comment (chengxiong-ruan) [reply pkg/sql/catalog/schematelemetry/schematelemetrycontroller/testdata/cron_rewrites:29]: heh, I think I looked at some bad oracle doc...

**Review by chengxiong-ruan — APPROVED** [review]

**Comment by ajwerner** [comment]
TFTR!

bors r+

**Review by postamar — APPROVED** [review]
LGTM.

Is any of this worth documenting? I don't think so but I'll ask anyway.

**Comment by ajwerner** [comment]
> Is any of this worth documenting?

I'm not particularly eager to. I vote we wait on that.

**Pull request merged**